From 05b3454da8679033cbb0f8acb99b436fc290af1d Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:33 -0400
Subject: [tools] perf/completion: Factor out compgen stuff

Message-id: <1408953210-26343-26-git-send-email-jolsa@redhat.com>
Patchwork-id: 88739
O-Subject: [PATCH RHEL7.1 BZ1133083 025/322] perf completion: Factor out compgen stuff
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 12f9dd5042483698c74a133d9004ff1d1a6474f9
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Sun Nov 17 21:43:24 2013 +0530

description
===========
compgen is a bash-builtin; factor out the invocations into a separate
function to give us a chance to override it with a zsh equivalent in
future patches.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index 3efdc84..8243126 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -89,6 +89,11 @@ __ltrim_colon_completions()
  fi
 }
 
+__perfcomp ()
+{
+ COMPREPLY=( $( compgen -W "$1" -- "$2" ) )
+}
+
 __perf_main ()
 {
  local cmd
@@ -99,23 +104,23 @@ __perf_main ()
  # List perf subcommands or long options
  if [ $cword -eq 1 ]; then
   if [[ $cur == --* ]]; then
-   COMPREPLY=( $( compgen -W '--help --version \
+   __perfcomp '--help --version \
    --exec-path --html-path --paginate --no-pager \
-   --perf-dir --work-tree --debugfs-dir' -- "$cur" ) )
+   --perf-dir --work-tree --debugfs-dir' -- "$cur"
   else
    cmds=$($cmd --list-cmds)
-   COMPREPLY=( $( compgen -W '$cmds' -- "$cur" ) )
+   __perfcomp "$cmds" "$cur"
   fi
  # List possible events for -e option
  elif [[ $prev == "-e" && "${words[1]}" == @(record|stat|top) ]]; then
   evts=$($cmd list --raw-dump)
-  COMPREPLY=( $( compgen -W '$evts' -- "$cur" ) )
+  __perfcomp "$evts" "$cur"
   __ltrim_colon_completions $cur
  # List long option names
  elif [[ $cur == --* ]];  then
   subcmd=${words[1]}
   opts=$($cmd $subcmd --list-opts)
-  COMPREPLY=( $( compgen -W '$opts' -- "$cur" ) )
+  __perfcomp "$opts" "$cur"
  fi
 }
 
-- 
1.7.1